ydb/core/tx/datashard/datashard__read_iterator.cpp

```diff
@@ -567,6 +573,7 @@ class TReader {
                FirstUnprocessedQuery++;
             else
                FirstUnprocessedQuery--;
+            LastProcessedKey.clear();
```

Comment:
Can you explain please when do we need to clear `LastProcessedKey`?
For an example I don't understand why doesn't it needed for `case EReadStatus::StoppedByLimit:`

Reply:
LastProcessedKey is used to resume iteration in a separate LocalDB transaction. When it's not set we start iteration as specified in the query. Since we move to the next query I clear LastProcessedKey to guarantee the next query doesn't use some unexpected last processed key from the previous query. Query iteration is supposed to set this key to the correct value before exiting, but conditions have become way too complicated over time, and it's better to be safe than sorry the next time.

When subquery returns StoppedByLimit the key to resume from has already been set to the correct value.